src/SFA.DAS.Activities/Extensions/ActivityExtensions.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.Activities.Models;
+using SFA.DAS.NLog.Logger;
+
+
+namespace SFA.DAS.Activities.Extensions
+{
+    public static class ActivityExtensions
+    {
+        private static readonly ILog Log = new NLogLogger(typeof(ActivityExtensions));
+
+        public static ActivityUrlLink GetLinkDetails(this Activity activity)
+        {
+            var link = activity.Type.GetActivityLink();
+
+            if (link == null)
+                return null;
+
+            if (activity.Type == ActivityType.PaymentCreated)
+            {
+                link.Parameters.Add("month", activity.At.Month);
+                link.Parameters.Add("year", activity.At.Year);
+            }
+
+            return link;
+        }
+
+        public static string GetMessageForActivity(this Activity activity, string message, params string[] dataItems)
+        {
+            return activity.GetMessageForActivity(message, null, dataItems);
```

Comment:
Consider providing an overload without the optional parameter rather than passing null (for formatter) as it simplifies the usage for the caller.

Reply:
This code was moved over from the ActivityType (was apart of the original code) and originally I didn't want to change anything regarding that code as it was just being moved to the correct extension class but i'll add in this as it makes sense anyway and won't increase the scope of the work too much.

Reply:
Having looked at this the method itself is the overload and extracting the logic out into a private method just pushing the issue along. As it is seems the least complex way to do it for now.